client/src/featureform/serving.py

```diff
@@ -653,8 +669,14 @@ def _register_model(
         self.db.insert("models", name, type)
         self.db.insert(look_up_table, name, association_name, association_variant)
 
-    def get_source_as_df(self, name, variant):
-        return self.get_input_df(name, variant)
+    def _get_source_as_df(self, name, variant, limit):
+        if limit == 0:
+            raise ValueError("limit must be greater than 0")
+        df = self.get_input_df(name, variant)
```

Comment:
shouldn't we pass the limit over to the server?

Reply:
This is the local implementation. We do pass it for the hosted implementation. See line 199 above.

Reply:
ohhhhh this is local client, got it